src/scan/postgres_scan.cpp

```diff
@@ -7,12 +9,70 @@
 #include "pgduckdb/pgduckdb_process_lock.hpp"
 #include "pgduckdb/logger.hpp"
 
+#include <numeric> // std::accumulate
+
 namespace pgduckdb {
 
 //
 // PostgresScanGlobalState
 //
 
+static duckdb::string
+FilterJoin(duckdb::vector<duckdb::string> &filters, duckdb::string &&delimiter) {
+	return std::accumulate(filters.begin() + 1, filters.end(), filters[0],
+	                       [&delimiter](duckdb::string l, duckdb::string r) { return l + delimiter + r; });
+}
+
+int
+PostgresScanGlobalState::ExtractQueryFilters(duckdb::TableFilter *filter, const char *column_name,
+                                             duckdb::string &query_filters, bool is_optional_filter_parent) {
```

Comment:
nit: "parent" seems like the wrong naming imo. Maybe one of these names instead:
```suggestion
                                             duckdb::string &query_filters, bool is_optional_filter_child) {
```

```suggestion
                                             duckdb::string &query_filters, bool is_inside_optional_filter) {
```